### PR TITLE
add pad

### PR DESCRIPTION
**Pull request by yeyeye333**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by yewentao256 — COMMENTED** [review]
需要删除device_config.py里的pad skip相关配置，将测试加入到ci中

**Comment by yeyeye333** [comment]
> 需要删除device_config.py里的pad skip相关配置，将测试加入到ci中

已修改

File: impl/camb/functions/pad.cpp [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+        int16_t tempI16 = 0;
+        uint16_t tempU16 = 0;
+        int32_t tempI32 = 0;
+        uint32_t tempU32 = 0;
+        int64_t tempI64 = 0;
+        uint64_t tempU64 = 0;
+        half_float::half tempF16 = static_cast<half_float::half>(0);
+        float tempF32 = 0;
+
+        if (value != nullptr) {
```
Comment (yewentao256) [inline impl/camb/functions/pad.cpp:105]: 这段逻辑有没有公共部分可以提取的？尽量不要都写在一个函数体里

File: impl/camb/functions/pad.cpp [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    DiopiTensor inputTensor(input);
+    DiopiTensor outTensor(out);
+    DiopiTensor inputTensorTmp = inputTensor;
+    DiopiTensor outTensorTmp = outTensor;
+
+    if (inputTensor.dtype() == diopi_dtype_float64) {
+        std::vector<DiopiTensor*> pTensors{&inputTensorTmp};
+        DIOPI_CALL(autoCastTensorType(ctx, pTensors, {diopi_dtype_float32}));
+        inputTensorTmp = *pTensors[0];
+        DIOPI_CALL(dataTypeCast(ctx, outTensorTmp, inputTensorTmp.dtype()));
```
Comment (yewentao256) [inline impl/camb/functions/pad.cpp:49]: 同之前的review，require tensor即可

Comment (yeyeye333) [reply impl/camb/functions/pad.cpp:49]: > 同之前的review，require tensor即可

已修改

File: impl/camb/functions/pad.cpp (hunk 1 above)
Comment (yeyeye333) [reply impl/camb/functions/pad.cpp]: > 这段逻辑有没有公共部分可以提取的？尽量不要都写在一个函数体里

已修改

**Review by yewentao256 — APPROVED** [review]

**Review by yangbofun — APPROVED** [review]

**Pull request merged**